components/rule/org.wso2.carbon.rule.backend/src/main/java/org/wso2/carbon/rule/backend/drools/DroolsBackendRuntime.java

```diff
@@ -76,67 +77,41 @@ public void addRuleSet(RuleSet ruleSet) throws RuleConfigurationException {
                             "Error in rule service configuration : Select \"regular\" as Resource Type for regular rules "
                                                                         + "or attached file is not supported by rule engine");
                 } else {
-                    DecisionTableConfiguration dtconf = KnowledgeBuilderFactory.newDecisionTableConfiguration();
-
-                    //check whether the decision tables is base on .xsl file of .csv format (inline input or .csv file input)
-                    if (rule.getSourceType().equalsIgnoreCase(Constants.RULE_SOURCE_TYPE_INLINE) ||
-                                                                    rule.getValue().lastIndexOf(".csv") > 0) {
-                        //decision table in .xls format
-                        dtconf.setInputType(DecisionTableInputType.CSV);
-                    } else {
-                        //decision table in .xls format
-                        dtconf.setInputType(DecisionTableInputType.XLS);
-                    }
-                    this.knowledgeBuilder.add(ResourceFactory.newInputStreamResource(ruleInputStream),
-                                              ResourceType.DTABLE, dtconf);
+                    this.kieFileSystem.write("src/main/resources/com/sample/"+ rule.hashCode() + ".drl",
+                            this.kieServices.getResources().newInputStreamResource(ruleInputStream));
                 }
 
             }
-
-            if (this.knowledgeBuilder.hasErrors()) {
-                throw new RuleConfigurationException("Error during creating rule set: " +
-                        this.knowledgeBuilder.getErrors());
-            }
-
-            Collection<KnowledgePackage> pkgs = this.knowledgeBuilder.getKnowledgePackages();
-            this.knowledgeBase.addKnowledgePackages(pkgs);
         }
 
-    }
-
-    public Session createSession(int type) throws RuleRuntimeException {
+        ReleaseId releaseId = this.kieServices.newReleaseId("com.sample", "my-sample-a", "1.0.0");
```

Comment:
Shall we revisit the name of the releaseID?

Reply:
fixed